server/service/osquery_utils/queries.go

```diff
@@ -393,23 +394,31 @@ SELECT
   'Browser plugin (Chrome)' AS type,
   '' AS bundle_identifier,
   'chrome_extensions' AS source
-FROM chrome_extensions
+FROM cached_users CROSS JOIN chrome_extensions USING (uid)
```

Comment:
Curious on why using the `CROSS` keyword (AFAICS it only provides allows some optimizations in `sqlite3` - https://www.sqlite.org/lang_select.html).

Reply:
Great question! There's some context in https://youtu.be/hcn3HIcHAAo?t=77, but what I didn't include there is that sometimes SQLite decides to reorder the nested loops for the table generation. `CROSS JOIN` tells SQLite that it must not reorder the tables, so we ensure that the `uid` parameter always gets passed to the generation function for the user-specific tables.

Reply:
Huh! We should add some comments for future readers, maybe just a link to the youtube lecture :)

Reply:
Hopefully this helps :) https://github.com/fleetdm/fleet/pull/3733/files#diff-9aebb07c5e9eae43b4cf167ef1ac12d720a613da219e5342d9a8e992e5fd0815R373-R377

Reply:
It does, appreciated!